rebasehelper/archive.py

```diff
@@ -82,134 +79,90 @@ def extract(cls, archive=None, filename=None, path=None):
 
 @register_archive_type
 class TarXzArchiveType(ArchiveTypeBase):
-
-    """ .tar.xz archive type """
-
     EXTENSION: str = '.tar.xz'
 
     @classmethod
-    def open(cls, filename=None):
-        if filename is None:
-            raise TypeError("Expected argument 'filename' (pos 1) is missing")
-        xz_file = lzma.LZMAFile(filename, "r")
-
-        return tarfile.open(mode='r', fileobj=xz_file)
+    def open(cls, filename):
+        return tarfile.open(mode='r', fileobj=lzma.LZMAFile(filename, 'r'))
 
     @classmethod
-    def extract(cls, archive=None, filename=None, path=None):
-        if archive is None:
-            raise TypeError("Expected argument 'archive' (pos 1) is missing")
+    def extract(cls, archive, filename, path):
         archive.extractall(path)
 
 
 @register_archive_type
-class Bz2ArchiveType(ArchiveTypeBase):
-
-    """ .bz2 archive type """
-
-    EXTENSION: str = '.bz2'
+class TarBz2ArchiveType(ArchiveTypeBase):
+    EXTENSION: str = '.tar.bz2'
 
     @classmethod
-    def open(cls, filename=None):
-        if filename is None:
-            raise TypeError("Expected argument 'filename' (pos 1) is missing")
-
-        if filename.endswith('.tar.bz2'):
+    def open(cls, filename):
+        if filename.endswith(TarBz2ArchiveType.EXTENSION):
```

Comment:
Is there any reason not to use `cls.EXTENSION`?

Reply:
Of course, the condition would be pointless in the case of `cls` being `Bz2ArchiveType`.

Reply:
Thanks for the explanation :)